clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -156,7 +177,8 @@ else if (LZ4.name.equals(name))
     private static class LZ4Constructors {
         static final MethodHandle INPUT = findConstructor(
                 "org.apache.kafka.common.record.KafkaLZ4BlockInputStream",
```

Comment:
note however that we store the LZ4SafeDecompressor as a static field, so loading the Kafka class would also trigger loaded the lz4 classes.